wren-core-py/src/lib.rs

```diff
@@ -1,120 +1,16 @@
-use std::sync::Arc;
-
-use base64::prelude::*;
 use pyo3::prelude::*;
 
-use crate::errors::CoreError;
-use crate::remote_functions::RemoteFunction;
-use log::debug;
-use wren_core::mdl;
-use wren_core::mdl::manifest::Manifest;
-use wren_core::mdl::AnalyzedWrenMDL;
+use remote_functions::PyRemoteFunction;
 
+pub mod context;
 mod errors;
-mod remote_functions;
-
-#[pyfunction]
-fn transform_sql(
-    mdl_base64: &str,
-    remote_functions: Vec<RemoteFunction>,
-    sql: &str,
-) -> Result<String, CoreError> {
-    let mdl_json_bytes = BASE64_STANDARD
-        .decode(mdl_base64)
-        .map_err(CoreError::from)?;
-    let mdl_json = String::from_utf8(mdl_json_bytes).map_err(CoreError::from)?;
-    let manifest = serde_json::from_str::<Manifest>(&mdl_json)?;
-    let remote_functions: Vec<mdl::function::RemoteFunction> = remote_functions
-        .into_iter()
-        .map(|f| f.into())
-        .collect::<Vec<_>>();
-
-    let Ok(analyzed_mdl) = AnalyzedWrenMDL::analyze(manifest) else {
-        return Err(CoreError::new("Failed to analyze manifest"));
-    };
-    match mdl::transform_sql(Arc::new(analyzed_mdl), &remote_functions, sql) {
-        Ok(transformed_sql) => Ok(transformed_sql),
-        Err(e) => Err(CoreError::new(&e.to_string())),
-    }
-}
-
-#[pyfunction]
-fn read_remote_function_list(path: Option<&str>) -> Vec<RemoteFunction> {
-    debug!(
-        "Reading remote function list from {}",
-        path.unwrap_or("path is not provided")
-    );
-    if let Some(path) = path {
-        csv::Reader::from_path(path)
-            .unwrap()
-            .into_deserialize::<RemoteFunction>()
-            .filter_map(Result::ok)
-            .collect::<Vec<_>>()
-    } else {
-        vec![]
-    }
-}
+pub mod remote_functions;
 
 #[pymodule]
 #[pyo3(name = "wren_core")]
 fn wren_core_wrapper(m: &Bound<'_, PyModule>) -> PyResult<()> {
     env_logger::init();
-    m.add_function(wrap_pyfunction!(transform_sql, m)?)?;
-    m.add_function(wrap_pyfunction!(read_remote_function_list, m)?)?;
+    m.add_class::<context::PySessionContext>()?;
+    m.add_class::<PyRemoteFunction>()?;
     Ok(())
 }
-
-#[cfg(test)]
```

Comment:
The pyclass of pyo3 has issues in the rust test. https://github.com/PyO3/pyo3/issues/340
Because this test duplicates with the python test, remove it and test it in python only.